libbeat/scripts/Makefile

```diff
@@ -455,3 +455,8 @@ release: mage
 .PHONY: snapshot
 snapshot: mage
 	@SNAPSHOT=true mage package
+
+ifdef NO_COLLECT
```

Comment:
Can you elaborate on this change how it compares to the previous option where it was always there and just empty? Overall I like it but would like to understand better why you changed it. Didn't fully get this from 3) in your PR description.

Reply:
Hi Nicolas,

I will try to explain it with an example:

Let us suppose we want to run the `make update` recipe from libbeat/scripts/Makefile directly. That means we are not including the libbeat/scripts/Makefile inside another Makefile but we are using it as it is (this is what is happening in our [commit](https://github.com/nfvsap/beats/blob/9666854b9859f0ee1c1e435800c0604b71a78d47/generator/beat/%7Bbeat%7D/Makefile#L22)) inside `make pre-setup` recipe.

In that case the libbeat/scripts/Makefile `make update` has a dependency to `collect` as you can see [here](https://github.com/elastic/beats/blob/master/libbeat/scripts/Makefile#L314).

The current master branch makes the assumption that the libbeat/scripts/Makefile is always included inside other Makefiles and that a collect recipe is declared inside them.

As a result of the above, in this example, the direct use of `make update` is going to fail because it cannot find collect recipe declared inside libbeat/scripts/Makefile.

What we propose here is the following:
1. Declare an empty collect recipe inside libbeat/scripts/Makefile in order to solve the above issue.
2. Keep the same default behavior for existing Beats (eg. [Heartbeat](https://github.com/elastic/beats/blob/master/heartbeat/Makefile#L11), [Metricbeat](https://github.com/elastic/beats/blob/master/metricbeat/Makefile#L52) etc.). That is: declare collect recipe if you are including libbeat/scripts/Makefile and using its update recipe.
3. Insert a variable NO_COLLECT in generated Beats. If the NO_COLLECT is set, then the empty `make collect` recipe is used from libbeat/scripts/Makefile.
4. If after generating a Beat and running initial `make setup`, the user wants to use collect, then he should declare a collect recipe inside the generated Beat's Makefile. After that, every time he runs `make update` his collect recipe is going to be used.

Reply:
Thanks for the detailed explanation ❤️ Make sense.